File: clients/client-dart/lib/src/json_decoder.dart
import 'dart:convert';

import 'channel_message.dart';
import 'message_decoder.dart';
import 'utils.dart';

class JsonDecoder extends MessageDecoder<String> {
  @override
  ChannelMessage decode(String event) {
    var eventAsList = jsonDecode('{"received": $event }')['received'];

    return ChannelMessage(
      Utils.checkString(eventAsList[0]),
      Utils.checkString(eventAsList[1]),
      Utils.checkString(eventAsList[2]),
      eventAsList[3],
    );
  }
}

File: clients/client-js/src/decoder/json-decoder.ts
/* The Json serializer for encoding and decoding messages */


import { ChannelMessage } from "../channel-message.js";
import { MessageDecoder } from "./message-decoder.js";

export class JsonDecoder implements MessageDecoder {

    public decode(messageEvent: MessageEvent): ChannelMessage {
        const [message_id, correlation_id, event, payload] = JSON.parse(messageEvent.data);
        return new ChannelMessage(message_id, event, correlation_id, payload);
    }

    public decode_sse(sse_event: string): ChannelMessage {
        const [message_id, correlation_id, event, payload] = JSON.parse(sse_event);
        return new ChannelMessage(message_id, event, correlation_id, payload);
    }
}

